monai/networks/nets/resnet.py

```diff
@@ -329,21 +339,40 @@ def _resnet(
     block: type[ResNetBlock | ResNetBottleneck],
     layers: list[int],
     block_inplanes: list[int],
-    pretrained: bool,
+    pretrained: bool | str,
     progress: bool,
     **kwargs: Any,
 ) -> ResNet:
     model: ResNet = ResNet(block, layers, block_inplanes, **kwargs)
     if pretrained:
-        # Author of paper zipped the state_dict on googledrive,
-        # so would need to download, unzip and read (2.8gb file for a ~150mb state dict).
-        # Would like to load dict from url but need somewhere to save the state dicts.
-        raise NotImplementedError(
-            "Currently not implemented. You need to manually download weights provided by the paper's author"
-            " and load then to the model with `state_dict`. See https://github.com/Tencent/MedicalNet"
-            "Please ensure you pass the appropriate `shortcut_type` and `bias_downsample` args. as specified"
-            "here: https://github.com/Tencent/MedicalNet/tree/18c8bb6cd564eb1b964bffef1f4c2283f1ae6e7b#update20190730"
-        )
+        if isinstance(pretrained, str):
+            if Path(pretrained).exists():
+                logger.info(f"Loading weights from {pretrained}...")
+                checkpoint = torch.load(pretrained, map_location=device)
+            else:
+                ### Throw error
+                raise FileNotFoundError("The pretrained checkpoint file is not found")
+        else:
+            resnet_depth = int(re.search(r"resnet(\d+)", arch).group(1))
+            # Download the MedicalNet pretrained model
+            logger.info(f"Loading MedicalNet pretrained model from https://huggingface.co/{MEDICALNET_HUGGINGFACE_REPO_BASENAME}{resnet_depth}")
+            pretrained_path = hf_hub_download(
+                repo_id=f"{MEDICALNET_HUGGINGFACE_REPO_BASENAME}{resnet_depth}",
+                filename=f"{MEDICALNET_HUGGINGFACE_FILES_BASENAME}{resnet_depth}.pth")
+
+            checkpoint = torch.load(pretrained_path, map_location=device)
+
+        if "state_dict" in checkpoint:
+            model_state_dict = checkpoint["state_dict"]
+            model_state_dict = {key.replace("module.", ""): value for key, value in model_state_dict.items()}
+        else:
+            ### Throw error
+            raise KeyError(
```

Comment:
May I know why must have this `state_dict` key here? What if users use their own pre-trained weights?

Reply:
This was to be consistent with the way weights are saved in MedicalNet (checkpoint containing the state dict). But it's probably best to pass the state dict directly when users pass their own pretrained weights ? 

```suggestion
            model_state_dict = checkpoint
```

Reply:
Yes, sure. You can demonstrate it in the docstring.